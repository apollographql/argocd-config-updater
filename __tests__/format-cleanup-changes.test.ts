import {
  formatCleanupChanges,
  CleanupChange,
} from '../src/format-cleanup-changes';

describe('formatCleanupChanges', () => {
  it('should return empty string for no changes', () => {
    const result = formatCleanupChanges([]);
    expect(result).toBe('');
  });

  it('should find single closed PRs', () => {
    const changes: CleanupChange[] = [
      {
        prNumber: 123,
        prTitle: 'Add new feature',
        prURL: 'https://github.com/owner/repo/pull/123',
      },
    ];

    const result = formatCleanupChanges(changes);
    expect(result).toContain('Found 1 closed PR:');
    expect(result).toContain(
      '- PR [#123](https://github.com/owner/repo/pull/123): Add new feature',
    );
  });

  it('should find multiple closed PRs', () => {
    const changes: CleanupChange[] = [
      {
        prNumber: 456,
        prTitle: 'Second PR',
        prURL: 'https://github.com/owner/repo/pull/456',
      },
      {
        prNumber: 123,
        prTitle: 'First PR',
        prURL: 'https://github.com/owner/repo/pull/123',
      },
    ];

    const result = formatCleanupChanges(changes);
    expect(result).toContain('Found 2 closed PRs:');
    expect(result).toContain(
      '- PR [#123](https://github.com/owner/repo/pull/123): First PR',
    );
    expect(result).toContain(
      '- PR [#456](https://github.com/owner/repo/pull/456): Second PR',
    );
  });

  it('should sort PRs by number', () => {
    const changes: CleanupChange[] = [
      {
        prNumber: 456,
        prTitle: 'Second PR',
        prURL: 'https://github.com/owner/repo/pull/456',
      },
      {
        prNumber: 123,
        prTitle: 'First PR',
        prURL: 'https://github.com/owner/repo/pull/123',
      },
    ];

    const result = formatCleanupChanges(changes);
    const lines = result.split('\n');
    const pr123Index = lines.findIndex((line) => line.includes('#123'));
    const pr456Index = lines.findIndex((line) => line.includes('#456'));
    expect(pr123Index).toBeLessThan(pr456Index);
  });
});
