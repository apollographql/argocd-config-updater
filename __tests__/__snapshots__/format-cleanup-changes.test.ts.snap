// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatCleanupChanges should format a single closed PR 1`] = `
"Found 1 closed PR:

**test-app** (dev)
- PR [#123](https://github.com/owner/repo/pull/123): Add new feature (closed Jan 15, 2024)"
`;

exports[`formatCleanupChanges should format multiple closed PRs under same app and environment 1`] = `
"Found 2 closed PRs:

**test-app** (dev)
- PR [#123](https://github.com/owner/repo/pull/123): First PR (closed Jan 15, 2024)
- PR [#456](https://github.com/owner/repo/pull/456): Second PR (closed Jan 18, 2024)"
`;

exports[`formatCleanupChanges should format single closed PR 1`] = `
"Found 1 closed PR:

**test-app** (dev)
- PR [#123](https://github.com/owner/repo/pull/123): Add new feature (closed Jan 15, 2024)"
`;

exports[`formatCleanupChanges should handle multiple PRs across different apps and environments 1`] = `
"Found 3 closed PRs:

**test-app** (dev)
- PR [#123](https://github.com/owner/repo/pull/123): Dev PR 1 (closed Jan 15, 2024)
- PR [#789](https://github.com/owner/repo/pull/789): Dev PR 2 (closed Jan 20, 2024)

**test-app** (staging)
- PR [#456](https://github.com/owner/repo/pull/456): Staging PR (closed Jan 18, 2024)"
`;

exports[`formatCleanupChanges should separate same app with different environments 1`] = `
"Found 2 closed PRs:

**test-app** (dev)
- PR [#123](https://github.com/owner/repo/pull/123): Dev PR (closed Jan 15, 2024)

**test-app** (staging)
- PR [#456](https://github.com/owner/repo/pull/456): Staging PR (closed Jan 18, 2024)"
`;

exports[`formatCleanupChanges should sort sections alphabetically by app name then environment 1`] = `
"Found 3 closed PRs:

**test-api** (dev)
- PR [#456](https://github.com/owner/repo/pull/456): API PR (closed Jan 18, 2024)

**test-app** (dev)
- PR [#789](https://github.com/owner/repo/pull/789): App Dev PR (closed Jan 20, 2024)

**test-app** (staging)
- PR [#123](https://github.com/owner/repo/pull/123): Router PR (closed Jan 15, 2024)"
`;
